src/Test/Utilities/Desktop/CompilerTraitAttribute.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Xunit;
+using Xunit.Sdk;
+
+namespace Microsoft.CodeAnalysis.Test.Utilities
+{
+    [TraitDiscoverer("Microsoft.CodeAnalysis.Test.Utilities.CompilerTraitDiscoverer", assemblyName: "Roslyn.Test.Utilities.Desktop")]
```

Comment:
There are a couple of places to read about:

- http://mac-blog.org.ua/xunit-category-trait/
- https://dennymichael.net/2015/01/23/how-to-extendmigrate-traitattribute-to-xunit-v2-and-why-has-become-sealed/

The purpose of the discoverer though is just to map the trait to a name / value pair.  This is how xunit fundamentally understands traits.  Their core TraitAttribute has the form `[Trait(name, value)]` and the discoverer just maps back the custom attribute to their understanding.

I have verified this implementation works correctly with the standard xunit libraries.